host/src/server.rs

```diff
@@ -71,11 +75,37 @@ fn set_headers(headers: &mut hyper::HeaderMap, extended: bool) {
 }
 
 #[derive(Clone)]
-struct Handler;
+struct Handler {
+    cache_dir: Option<PathBuf>,
+}
 
 impl Handler {
-    fn new() -> Self {
-        Self {}
+    pub fn new() -> Self {
+        Self { cache_dir: None }
+    }
+
+    pub fn new_with_cache(dir: PathBuf) -> Self {
+        Self {
+            cache_dir: Some(dir),
+        }
+    }
+
+    pub fn get(&self, block_no: u64, network: &str) -> Option<GuestInput> {
+        let mut input: Option<GuestInput> = None;
+        self.cache_dir
+            .as_ref()
+            .map(|dir| dir.join(format!("input-{}-{}", network, block_no)))
```

Comment:
could extract out the filename creation to a separate function so it's for sure consistent in both `get` and `set`.

Reply:
But this is already in the `get` & `set` function of `Handler`

Reply:
I mean just the filename generation, now the code depends on `get` and `set` to have matching filenames so they have to manually be kept consistent.